integration/spark2/src/main/scala/org/apache/spark/sql/events/MergeBloomIndexEventListener.scala

```diff
@@ -24,59 +24,96 @@ import org.apache.spark.internal.Logging
+        val bloomDatamaps = DataMapStoreManager.getInstance().getAllDataMap(carbonTable).asScala
+          .filter(_.getDataMapSchema.getProviderName.equalsIgnoreCase(
             DataMapClassProvider.BLOOMFILTER.getShortName))
-
-        if (datamapPostEvent.isFromRebuild) {
-          if (null != datamapPostEvent.dmName) {
-            // for rebuild process
-            bloomDatamaps = bloomDatamaps.filter(
-              _.getDataMapSchema.getDataMapName.equalsIgnoreCase(datamapPostEvent.dmName))
-          }
-        } else {
-          // for load process, skip lazy datamap
-          bloomDatamaps = bloomDatamaps.filter(!_.getDataMapSchema.isLazy)
-        }
+          .filter(_.getDataMapSchema.getDataMapName.equalsIgnoreCase(datamapPostEvent.dmName))
+          .toList
 
         val segmentIds = datamapPostEvent.segmentIdList
-        if (bloomDatamaps.size > 0 && segmentIds.size > 0) {
-          // we extract bloom datamap name and index columns here
-          // because TableDataMap is not serializable
-          val bloomDMnames = ListBuffer.empty[String]
-          val bloomIndexColumns = ListBuffer.empty[Seq[String]]
-          bloomDatamaps.foreach( dm => {
-            bloomDMnames += dm.getDataMapSchema.getDataMapName
-            bloomIndexColumns += dm.getDataMapSchema.getIndexColumns.map(_.trim.toLowerCase)
-          })
-          new CarbonMergeBloomIndexFilesRDD(sparkSession, carbonTable,
-            segmentIds, bloomDMnames, bloomIndexColumns).collect()
-        }
+        mergeBloomIndex(sparkSession, carbonTable, bloomDatamaps, segmentIds)
     }
   }
 
 
-  private def clearBloomCache(carbonTable: CarbonTable, segmentIds: Seq[String]): Unit = {
-    DataMapStoreManager.getInstance.clearDataMaps(carbonTable.getTableUniqueName)
+  private def mergeBloomIndex(sparkSession: SparkSession, carbonTable: CarbonTable,
+      bloomDatamaps: List[TableDataMap], segmentIds: Seq[String]) = {
+    if (bloomDatamaps.nonEmpty && segmentIds.nonEmpty) {
+      // we extract bloom datamap name and index columns here
+      // because TableDataMap is not serializable
+      val bloomDMnames = ListBuffer.empty[String]
+      val bloomIndexColumns = ListBuffer.empty[Seq[String]]
+      bloomDatamaps.foreach(dm => {
+        bloomDMnames += dm.getDataMapSchema.getDataMapName
+        bloomIndexColumns += dm.getDataMapSchema.getIndexColumns.map(_.trim.toLowerCase)
+      })
+      LOGGER.info(
+        String.format("Start to merge bloom index file for table %s. Datamaps=%s, SegmentIds=%s",
+          carbonTable.getTableName, bloomDMnames.mkString("|"), segmentIds.mkString("|") ))
+      new CarbonMergeBloomIndexFilesRDD(sparkSession, carbonTable,
+        segmentIds, bloomDMnames, bloomIndexColumns).collect()
+      LOGGER.info("Finish merging bloom index file for table " + carbonTable.getTableName)
+    } else {
+      LOGGER.info("No segment of any bloom datamap needs to merge bloom index file for table "
```

Comment:
I think we can just skip this ELSE branch. For normal table without bloomdatamap, we do not need to bother the user with this information.

Reply:
OK, I'll remove this